source/slang/slang-lookup.cpp

```diff
@@ -224,14 +224,18 @@ static void _lookUpDirectAndTransparentMembers(
         }
     }
 
-    // TODO(tfoley): should we look up in the transparent decls
-    // if we already has a hit in the current container?
     for(auto transparentInfo : containerDecl->getTransparentMembers())
     {
         // The reference to the transparent member should use the same
         // path as we used in referring to its parent.
         DeclRef<Decl> transparentMemberDeclRef = astBuilder->getMemberDeclRef(parentDeclRef, transparentInfo.decl);
 
+        // If we have already started checking this member, there is no need to send another check.
```

Comment:
Why do we need to change the lookup logic to prevent the diagnostics? It seems like lookup shouldn't be responsible for this?